pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/utils/DoubleVectorOpUtils.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.utils;
+
+import com.google.common.base.Preconditions;
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+
+
+public class DoubleVectorOpUtils {
+  private DoubleVectorOpUtils() {
+  }
+
+  public static DoubleArrayList vectorAdd(DoubleArrayList a, DoubleArrayList b) {
+    Preconditions.checkState(a.size() == b.size(), "The two operand arrays are not of the same size! provided %s, %s",
+        a.size(), b.size());
+    for (int i = 0; i < a.size(); i++) {
+      a.elements()[i] += b.elements()[i];
+    }
+    return a;
+  }
+
+  public static DoubleArrayList vectorAdd(DoubleArrayList a, double[] b) {
+    Preconditions.checkState(a.size() == b.length, "The two operand arrays are not of the same size! provided %s, %s",
```

Comment:
Call `a.elements()` first and cache the value, then operate on 2 `double[]` for better performance. Not sure if jvm will inline all the method calls. Same for `incrementElement()`